app/logic/Auth/OAuth2.ts

```diff
@@ -182,16 +184,27 @@ export class OAuth2 extends Observable {
    * @throws OAuth2Error
    */
   protected async getAccessTokenFromParams(params: any, additionalHeaders?: any, tokenURL: string | void = this.tokenURL): Promise<string> {
-    params.scope = this.scope;
     params.client_id = this.clientID;
-    params.client_secret = this.clientSecret || undefined;
+    
+    if (this.clientSecret) {
```

Comment:
Why this criteria to turn on/off PKCE?
Can we support a) client secret with PKCE and b) no client secret and no PKCE ?
So far, we use case b), so it seems like case that should be possible.

Reply:
Yes, we could do that. How would you like to turn on/off PKCE? With a property PKCE?

Reply:
Yes, it should be a property `doPKCE` on the OAuth2 object. We could then set it from the OAuth2URLs list.